static/js/components/JobDetails.jsx

```diff
@@ -0,0 +1,218 @@
+var AddForm = React.createClass({
+
+  getInitialState: function() {
+    return {clicked: false, errored: false};
+  },
+
+  formSubmit: function(e) {
+    e.preventDefault();
+    var crontime = this.refs.crontime.getInputDOMNode().value;
+    var workload = this.refs.workload.getInputDOMNode().value;
+    $.ajax({
+      url: "/jobs",
+      type: "POST",
+      data: {Function: this.props.function, CronTime: crontime, Workload: workload},
+      dataType: "json",
+      success: function(data) {
+        this.props.getJobDetails(this.props.function);
+      }.bind(this),
+      error: function(xhr, status, err) {
+        this.setState({errored: true, err_msg: xhr.responseText});
+      }.bind(this)
+    }); 
+  },
+
+  addJob: function() {
+    this.setState({clicked: true});
+  },
+
+  cronAlert: function() {
+    if (!this.state.errored) {
+      return;
+    }
+    return <Alert bsStyle="danger"> {this.state.err_msg} </Alert>;
+  },
+
+  render: function() {
+    if (!this.state.clicked) {
+      return (<div><Button bsStyle="primary" onClick={this.addJob}> Add job </Button><p></p></div>);
+    }
+
+    var crontime_placeholder = 'Cron time: (e.g.  0 13 */4 * * 1-5)';
+    var workload_placeholder = 'Workload: (e.g. "--task=job" or {"task":"job"})';
+    return (
+      <div>
+        {this.cronAlert()}
+        <form onSubmit={this.formSubmit} method="POST">
+        <Input ref="crontime" type="text" placeholder={crontime_placeholder} required />
+        <Input ref="workload" type="text" placeholder={workload_placeholder}/>
+        <ButtonInput bsStyle="primary" type="submit">Submit job</ButtonInput>
+        </form>
+      </div>
+    );
+  }
+});
+
+var CronRow = React.createClass({
+  getInitialState: function() {
+    return {button_clicked: false, delete_clicked: false};
+  },
+
+  button_click: function() {
```

Comment:
Can you give this a more descriptive name (also it's dependents). Perhaps `toggle_activated_click`?